microsoft/testsuites/xfstests/xfstesting.py

```diff
@@ -55,39 +106,90 @@ def _prepare_data_disk(
-        version = "3.1.1"
-    else:
-        version = "3.0"
-    return version
+# DEPRECATED !!!
+# This does not works on newer kernels.
+# Pls see:
+# https://lists.samba.org/archive/samba-technical/2018-June/128806.html
+# def _get_smb_version(node: Node) -> str:
+#     if node.tools[KernelConfig].is_enabled("CONFIG_CIFS_SMB311"):
+#         version = "3.1.1"
+#     else:
+#         version = "3.0"
+#     return version
 
 
-def _prepare_azure_file_share(
+def _deploy_azure_file_share(
     node: Node,
-    account_credential: Dict[str, str],
-    test_folders_share_dict: Dict[str, str],
-    fstab_info: str,
-) -> None:
-    folder_path = node.get_pure_path("/etc/smbcredentials")
-    if node.shell.exists(folder_path):
-        node.execute(f"rm -rf {folder_path}", sudo=True)
-    node.shell.mkdir(folder_path)
-    file_path = node.get_pure_path("/etc/smbcredentials/lisa.cred")
-    echo = node.tools[Echo]
-    username = account_credential["account_name"]
-    password = account_credential["account_key"]
-    echo.write_to_file(f"username={username}", file_path, sudo=True, append=True)
-    echo.write_to_file(f"password={password}", file_path, sudo=True, append=True)
-    node.execute("cp -f /etc/fstab /etc/fstab_cifs", sudo=True)
-    for folder_name, share in test_folders_share_dict.items():
-        node.execute(f"mkdir {folder_name}", sudo=True)
-        echo.write_to_file(
-            f"{share} {folder_name} cifs {fstab_info}",
-            node.get_pure_path("/etc/fstab"),
-            sudo=True,
-            append=True,
+    environment: Environment,
+    file_share_name: str,
+    scratch_name: str,
+    azure_file_share: Any,
+    allow_shared_key_access: bool = True,
+    enable_private_endpoint: bool = True,
+    storage_account_sku: str = "Standard_LRS",
+    storage_account_kind: str = "StorageV2",
+    file_share_protocol: str = "SMB",
+    file_share_quota_in_gb: int = 500,
+) -> Dict[str, str]:
+    """
+    About: This method will provision azure file shares on a new // existing
+    storage account.
+    Returns: Dict[str, str] - A dictionary containing the file share names
+    and their respective URLs.
+    """
+    if isinstance(azure_file_share, AzureFileShare):
+        file_share_protocol = "SMB"
+    elif isinstance(azure_file_share, Nfs):
+        file_share_protocol = "NFS"
+    else:
+        raise LisaException("Unsupported file share protocol")
+    if file_share_protocol == "SMB":
+        fs_url_dict: Dict[str, str] = azure_file_share.create_file_share(
+            file_share_names=[file_share_name, scratch_name],
+            environment=environment,
+            sku=storage_account_sku,
+            kind=storage_account_kind,
+            allow_shared_key_access=allow_shared_key_access,
+            enable_private_endpoint=enable_private_endpoint,
+            quota_in_gb=file_share_quota_in_gb,
         )
+        test_folders_share_dict = {
+            _test_folder: fs_url_dict[file_share_name],
```

Comment:
Define a class, instead of use the dict. It's easy to misuse the dict.

Reply:
Follow previous code, pass in the names as a list like `[file_share_name, scratch_name]`, and then return a dict with their name/values. It saves code, and not hardcode them.